src/common/transformations/src/transformations/common_optimizations/concat_to_broadcast.cpp

```diff
@@ -0,0 +1,63 @@
+// Copyright (C) 2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+// #include "/transformations/common_optimizations/concat_to_broadcast.hpp"
+#include "transformations/common_optimizations/concat_to_broadcast.hpp"
+
+#include "itt.hpp"
+#include "openvino/pass/pattern/op/wrap_type.hpp"
+#include "openvino/pass/graph_rewrite.hpp"
+#include "transformations/utils/utils.hpp"
+#include "openvino/op/concat.hpp"
+#include "openvino/op/broadcast.hpp"
+
+ov::pass::ConcatToBroadcast::ConcatToBroadcast() {
+    MATCHER_SCOPE(ConcatToBroadcast);
+
+    auto concat_label = pattern::wrap_type<op::v0::Concat>([](const Output<Node>& value) {
+        auto node = value.get_node_shared_ptr();
+        auto first_input_node = node->get_input_source_output(0).get_node_shared_ptr();
+
+        for (const auto& input : node->input_values()) { // TODO: don't start from 0 to not do an additional comparison
+            if (input.get_node_shared_ptr() != first_input_node) {
+                return false;
+            }
+        }
+
+        return true;
+    });
+
+    matcher_pass_callback callback = [OV_CAPTURE_CPY_AND_THIS](pattern::Matcher& m) {
+        std::cout << "______" << matcher_name << " start______" << std::endl;
+        const auto& pattern_map = m.get_pattern_value_map();
+
+        auto concat = pattern_map.at(concat_label);
+
+        auto input = concat.get_node_shared_ptr()->input_value(0).get_node_shared_ptr(); // Very sure this is incorrect
```

Comment:
it's not correct
auto input = concat.get_node_shared_ptr()->input_value(0) is enough